### ISSUE TITLE
isinstance(x, Tuple) fails when equivalent expressions don't

### ISSUE THREAD

**Issue by lbenezriravin** [python/mypy#6680]
**reproducer**
```
from typing import Any, Dict, Tuple

x = (1,2)
isinstance(x, tuple)
isinstance(x, Tuple[Any, ...])
isinstance(x, Tuple)
```
**observed behavior**
```
$ pipenv run mypy test.py 
test.py:6: error: Argument 2 to "isinstance" has incompatible type "_SpecialForm"; expected "Union[type, Tuple[Union[type, Tuple[Any, ...]], ...]]"
```

**expected behavior**
[The python documentation states](https://docs.python.org/3/library/typing.html#typing.Tuple)
> A plain Tuple is equivalent to Tuple[Any, ...], and in turn to tuple.

If the above is true, IMO either all of the examples should raise errors, or none of them should. Only the third example is raising an error right now. I'm not quite sure what the error means in any case.

**environment**
master as of 2019-04-15
```
$ pipenv graph
mypy==0.710+dev.2c8a809693cf5be0b29d272d7f0711521bd4a55a
  - mypy-extensions [required: >=0.4.0,<0.5.0, installed: 0.4.1]
  - typed-ast [required: >=1.3.1,<1.4.0, installed: 1.3.1]
```

As always, thanks for all the work you do here!

**Comment by gvanrossum** [comment]
I am guessing the sentence you quote from the docs applies to its use in type annotations, not in `isinstance()`. In that case IMO *only* `tuple` should be allowed, because at *runtime* the *other* use of `Tuple` raises a `TypeError`. IMO it's a mypy bug that it doesn't report both uses of `Tuple` as errors. (I'm too lazy to look up what the *mypy* docs say about this -- they tend to be more correct than the Python docs about type checking matters.)

**Comment by lbenezriravin** [comment]
Ah, I see that now. If you don't mind me asking, why does `Tuple` behave differently from other superficially similar generics? For example,
```
y = {1: 1}
isinstance(y, Dict)
isinstance(y, Sequence)
```
don't raise TypeErrors unless I parameterize the generics.

**Comment by gvanrossum** [comment]
Because `Tuple` is not a generic class, it's a special form. This is so that it can take a variable number of parameters.

**Comment by lbenezriravin** [comment]
That makes sense. Thanks for the guidance!

**Comment by ilevkivskyi** [comment]
Essentially, the mypy behavior is opposite to what happens at runtime:
```python
isinstance((), Tuple)  # Works at runtime, error in mypy
isinstance((), Tuple[Any, ...])  # Error at runtime, no error in mypy
```
I think however this is a low priority, you should really just use `tuple`.

**Comment by AlexWaygood** [comment]
On 0.941, mypy now correctly [emits an error](https://mypy-play.net/?mypy=latest&python=3.10&gist=2127045f0518c353365f8d1af237baa4) for `isinstance(x, Tuple[Any, ...])`. It still gives a false-positive error for `isinstance(x, Tuple)`, though:

```python
from typing import Any, Dict, Tuple

x = (1,2)
isinstance(x, tuple)
isinstance(x, Tuple[Any, ...])  # error: Argument 2 to "isinstance" has incompatible type "object"; expected "Union[type, UnionType, Tuple[Union[type, UnionType, Tuple[Any, ...]], ...]]"
isinstance(x, Tuple)  # error: Argument 2 to "isinstance" has incompatible type "_SpecialForm"; expected "Union[type, UnionType, Tuple[Union[type, UnionType, Tuple[Any, ...]], ...]]"
```

**Comment by AlexWaygood** [comment]
(Closed by accident)

**Comment by gvanrossum** [comment]
So what's to do here? The OP is about runtime behavior, which should not be discussed in the mypy tracker.

**Comment by AlexWaygood** [comment]
I read the OP as being about the false-positive error mypy emits on this line of code:

```python
isinstance(x, Tuple)
```

This works fine at runtime, but mypy 0.942 reports:

```
error: Argument 2 to "isinstance" has incompatible type "_SpecialForm"; expected "Union[type, UnionType, Tuple[Union[type, UnionType, Tuple[Any, ...]], ...]]"
```

It's hardly a priority to fix, but it does seem to me to be a legitimate bug?

**Comment by AlexWaygood** [comment]
I've just closed #6864 as a duplicate -- #6864 is about `isinstance(x, collections.abc.Callable)` causing mypy to error out, with exactly the same error message. (That also works fine at runtime.)

**Comment by randolf-scholz** [comment]
https://github.com/python/mypy/issues/14014 is likely also a duplicate. This issue persists when checking against `Callable` class pattern in match-case statements

```python
match x:
    case collections.abc.Callable() as func:
        ...
```

Literally all other classes in `collections.abc` seem to work without problem.
